File: src/main/java/com/example/khuasenov/fluxstarter/repository/PersonRepository.java
package com.example.khuasenov.fluxstarter.repository;

import com.example.khuasenov.fluxstarter.model.Persons;
import java.util.List;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

/**
 * @author Khusaenov on 17.07.2018
 */
@Repository
public interface PersonRepository extends MongoRepository<Persons, String> {


    public List<Persons> findPersonByName();

    public List<Persons> findAll();

    public List<Persons> findByAge();


}

File: src/main/kotlin/com/example/khusaenov/fluxstarter/repository/PersonRepository.kt
package com.example.khusaenov.fluxstarter.repository

import com.example.khusaenov.fluxstarter.model.Persons
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

/**
 * @author Khusaenov on 10.08.2018
 */
interface PersonRepository : ReactiveMongoRepository<Persons, String> {

    fun findPersonByName(): Flux<Persons>

    override fun findAll(): Flux<Persons>

    fun findByAge(): Flux<Persons>
}